File: front/hackaton/src/app/Produto.ts
import { Fornecedor } from "./Fornecedor"
import { Negociacao } from "./Negociacao"
export interface Produto {
  produtoId: number,
  nomeProduto: string,
  valorUnitario: number,
  quantidadeProduto: number,
  descricaoProduto: string,
  fornecedor: Fornecedor,
  negociacao: Negociacao
}

File: Backend/comercioLocalAPI/Model/Produto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace comercioLocalAPI.Model
{
    public class Produto
    {
        public int produtoId { get; set; }
        public string? nomeProduto { get; set; }
        public double valorUnitario { get; set; }
        public int quantidadeProduto { get; set; }
        public string? descricaoProduto { get; set; }
        
        public Fornecedor fornecedor { get;} = new();
        public Negociacao negociacao { get;} = new();
    }
}